hasher-matcher-actioner/webapp/src/Api.tsx

```diff
@@ -324,38 +326,83 @@ export async function fetchHashCount(): Promise<Response> {
 
 type AllActions = {
   error_message: string;
-  actions_response: Array<any>;
+  actions_response: Array<{
+    name: string;
+    config_subtype: string;
+    url: string;
+    headers: string;
+  }>;
 };
 
-export async function fetchAllActions(): Promise<AllActions> {
-  return apiGet('actions/');
+export async function fetchAllActions(): Promise<Action[]> {
+  return apiGet<AllActions>('actions/').then(response => {
+    if (response && !response.error_message && response.actions_response) {
+      return response.actions_response.map(
+        ({name, config_subtype, url, headers}) =>
+          new Action(name, config_subtype, url, headers),
+      );
+    }
+    return [];
+  });
 }
 
 export async function createAction(
-  newAction = {},
+  newAction: Action,
 ): Promise<{response: string}> {
-  return apiPost('actions/', newAction);
+  return apiPost('actions/', {
+    name: newAction.name,
+    config_subtype: newAction.config_subtype,
+    fields: newAction.params,
+  });
 }
 
 export async function updateAction(
-  name: string,
-  type: string,
-  updatedAction = {},
+  old_name: string,
+  old_config_subtype: string,
+  updatedAction: Action,
 ): Promise<{response: string}> {
-  return apiPut(`actions/${name}/${type}`, updatedAction);
+  return apiPut(`actions/${old_name}/${old_config_subtype}`, {
+    name: updatedAction.name,
+    config_subtype: updatedAction.config_subtype,
+    fields: updatedAction.params,
+  });
 }
 
 export async function deleteAction(name: string): Promise<{response: string}> {
   return apiDelete(`actions/${name}`);
 }
 
+type APIActionRule = {
```

Comment:
We need two different ActionRule types because the Backend model is different from the front end model

Reply:
We could (and probably should) change this in the future but I would rather first type the current model and then change it

Reply:
nit: if you feel the need to add this clarification/context, why not add it as a comment? Especially if you think it is worth of a ToDo?

Reply:
Agreed

Reply:
did I miss you adding this somewhere?

Reply:
Nope. Just waiting to adress all your changes at once